File: backend/app/Models/Materiel.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

#[Fillable(['nom', 'categorie', 'description', 'quantite_disponible'])]
class Materiel extends Model
{
    use HasFactory;

    public function demandeMateriels(): HasMany
    {
        return $this->hasMany(DemandeMateriel::class, 'materiel_id');
    }
}

File: mobile/lib/models/materiel.dart
class Materiel {
  final int id;
  final String nom;
  final String? categorie;
  final String? description;
  final int quantiteDisponible;

  Materiel({
    required this.id,
    required this.nom,
    required this.categorie,
    required this.description,
    required this.quantiteDisponible,
  });

  factory Materiel.fromJson(Map<String, dynamic> json) {
    return Materiel(
      id: json['id'] as int,
      nom: json['nom'] as String,
      categorie: json['categorie'] as String?,
      description: json['description'] as String?,
      quantiteDisponible: json['quantite_disponible'] as int,
    );
  }
}
